squire_web/src/tournament/rounds/selected.rs

```diff
@@ -20,15 +20,40 @@ impl SelectedRound {
     }
 
     pub fn view(&self, tourn: &Tournament) -> Html {
-        html! { <p>{ self
+        let returnhtml = self
             .id
             .map(|id| {
                 tourn
                     .get_round(&id.into())
-                    .map(|rnd| format!("Round #{} at table #{}", rnd.match_number, rnd.table_number))
-                    .unwrap_or_else(|_| "Round not found!!!".to_owned())
+                    .map(|rnd| {
+                        html! {
+                            <>
+                            <h4>{ format!("Round #{} at table #{}", rnd.match_number, rnd.table_number) }</h4>
+                            <p>{ format!("# of players : {}", rnd.players.len()) }</p>
+                            <p>{ format!("Active : {}", rnd.is_active()) }</p>
+                            <p>{ format!("Bye : {}", rnd.is_bye()) }</p>
+                            <p>{ format!("Players : {}", rnd.players.len() ) }</p>
+                            <ul>
+                            {
+                                rnd.players.clone().into_iter()
+                                    .map(|pid| {
+                                        html! { <li>{ format!( "{}", tourn.get_player(&pid.into()).map(|p| p.name.as_str()).unwrap_or_else(|_| "Player not found") ) }</li>}
```

Comment:
Same issue here with the length of the function chains. However, I like that you're avoiding extra allocations by using `String::as_str()`. It's small, but every bit helps the performance of the UI.